errors.go

```diff
@@ -81,8 +82,19 @@ func (p *Error) Format(f fmt.State, c rune) {
 
 // LogMetadata implements the logMetadataProvider interface in the slog library which means that
 // the error params will automatically be merged with the slog metadata.
+// Additionally we put stack data in here for slog use.
 func (p *Error) LogMetadata() map[string]string {
-	return p.Params
+	logParams := map[string]string{
+		"terrors_file": p.StackFrames[0].Filename,
+		"terrors_function": p.StackFrames[0].Method,
+		"terrors_line":strconv.Itoa(p.StackFrames[0].Line),
```

Comment:
I think we should be tolerant of the stack frames being empty and do a `len` check.